--- texinfo-4.13/doc/texinfo.txi.xz~	2008-12-26 13:42:41.668546075 +0100
+++ texinfo-4.13/doc/texinfo.txi	2008-12-26 13:43:42.662523806 +0100
@@ -16488,6 +16488,7 @@ If @var{dir-file} (however specified) do
 
 @cindex Compressed dir files, reading
 @cindex Bzipped dir files, reading
+@cindex XZ-compressed dir files, reading
 @cindex LZMA-compressed dir files, reading
 @cindex Dir files, compressed
 If any input file is compressed with @code{gzip} (@pxref{Top,,,gzip,
@@ -16495,8 +16496,8 @@ Gzip}), @code{install-info} automaticall
 for reading.  And if @var{dir-file} is compressed, @code{install-info}
 also automatically leaves it compressed after writing any changes.
 If @var{dir-file} itself does not exist, @code{install-info} tries to
-open @file{@var{dir-file}.gz}, @file{@var{dir-file}.bz2}, and 
-@file{@var{dir-file}.lzma}, in that order.
+open @file{@var{dir-file}.gz}, @file{@var{dir-file}.bz2},
+@file{@var{dir-file}.xz} and @file{@var{dir-file}.lzma}, in that order.
 
 Options:
 
--- texinfo-4.13/info/filesys.c.xz~	2008-12-26 13:30:56.628544734 +0100
+++ texinfo-4.13/info/filesys.c	2008-12-26 13:44:16.108524823 +0100
@@ -55,6 +55,7 @@ static char *info_suffixes[] = {
 static COMPRESSION_ALIST compress_suffixes[] = {
   { ".gz", "gunzip" },
   { ".bz2", "bunzip2" },
+  { ".xz", "unxz" },
   { ".lzma", "unlzma" },
   { ".z", "gunzip" },
   { ".Z", "uncompress" },
--- texinfo-4.13/install-info/install-info.c.xz~	2008-12-26 13:31:20.607545485 +0100
+++ texinfo-4.13/install-info/install-info.c	2008-12-26 13:44:32.878774674 +0100
@@ -739,15 +739,20 @@ open_possibly_compressed_file (char *fil
 #else
     *compression_program = "bzip";
 #endif
+  else if (data[0] == '\xFD' && data[1] == '7' && data[2] == 'z'
+               && data[3] == 'X' && data[4] == 'Z' && data[5] == 0x00)
+#ifndef STRIP_DOT_EXE
+    *compression_program = "xz.exe";
+#else
+    *compression_program = "xz";
+#endif
     /* We (try to) match against old lzma format (which lacks proper
-       header, two first matches), as well as the new format (last match).  */
+       header, two first matches). */
   else if ((data[9] == 0x00 && data[10] == 0x00 && data[11] == 0x00
             && data[12] == 0x00)
            || (data[5] == '\xFF' && data[6] == '\xFF' && data[7] == '\xFF'
                && data[8] == '\xFF' && data[9] == '\xFF' && data[10] == '\xFF'
-               && data[11] == '\xFF' && data[12] == '\xFF') 
-           || (data[0] == '\xFF' && data[1] == 'L' && data[2] == 'Z'
-               && data[3] == 'M' && data[4] == 'A' && data[5] == 0x00))
+               && data[11] == '\xFF' && data[12] == '\xFF'))
 #ifndef STRIP_DOT_EXE
     *compression_program = "lzma.exe";
 #else
